### ISSUE TITLE
cargo check failure

### ISSUE THREAD

**Issue by iitalics** [ancwrd1/p12-keystore#1]
cargo check fails on this crate:

```
error[E0446]: crate-private type `ParsedAuthSafe` in public interface
  --> src/codec.rs:91:1
   |
52 | pub(crate) struct ParsedAuthSafe {
   | -------------------------------- `ParsedAuthSafe` declared as crate-private
...
91 | pub fn parse_auth_safe(safe: &ContentInfo, password: &str) -> Result<ParsedAuthSafe> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private type

For more information about this error, try `rustc --explain E0446`.
error: could not compile `p12-keystore` (lib) due to previous error
```

i am getting this error due to this crate being a transitive dependency

**Comment by ancwrd1** [comment]
Which Rust version are you using? As you can see from the badge all CI checks are passing.

**Comment by iitalics** [comment]
using rust `1.73.0`. on further inspection it seems like behavior was changed in here https://rust-lang.github.io/rfcs/2145-type-privacy.html and that's why the function is now accepted.

would it be possible to patch this to simplify the visibility annotations and preserve backwards compatibility? i believe all that would be required is to change `pub(crate) mod` to `mod` and `pub(crate) struct` to `pub struct` https://gist.github.com/iitalics/82fbeea5eb90829ffd7e15d4689e866c

**Comment by ancwrd1** [comment]
Yeah let me fix it.

**Comment by ancwrd1** [comment]
Please confirm that it is fixed and I will publish it.

**Comment by iitalics** [comment]
it looks like this fixes it, thanks so much for your help!

**Comment by ancwrd1** [comment]
Ok, published version 0.1.3.
